sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -27,24 +61,21 @@ Azure::Core::Credentials::AccessToken ClientSecretCredential::GetToken(
   static std::string const errorMsgPrefix("ClientSecretCredential::GetToken: ");
   try
   {
+    auto const isAdfs = m_tenantId == "adfs";
+
     Url url(m_options.AuthorityHost);
     url.AppendPath(m_tenantId);
-    url.AppendPath("oauth2/v2.0/token");
+    url.AppendPath(isAdfs ? "oauth2/token" : "oauth2/v2.0/token");
 
     std::ostringstream body;
     body << "grant_type=client_credentials&client_id=" << Url::Encode(m_clientId)
          << "&client_secret=" << Url::Encode(m_clientSecret);
 
-    auto const& scopes = tokenRequestContext.Scopes;
-    if (!scopes.empty())
     {
-      auto scopesIter = scopes.begin();
-      body << "&scope=" << Url::Encode(*scopesIter);
-
-      auto const scopesEnd = scopes.end();
-      for (++scopesIter; scopesIter != scopesEnd; ++scopesIter)
+      auto const& scopes = tokenRequestContext.Scopes;
```

Comment:
Formatting looks off here, Might be github review though.

Reply:
If formatting was off, CI wouldfail. Here, indentation is correct. I added extra `{}` to localize `scopes`.

Reply:
Test cases will be added by #1596.
Right now, there is no infrastructure to mock web responses. The effort to add one testcase would result in more files changing, after which I will be asked to split this PR into multiple parts - this original PR, and the other fixing just the #1596.

Reply:
We seem to be accruing a lot of test debt against that issue. Either prioritize that issue and get the test structure in place, first, OR for new code being checked-in add some unit tests. We need to ensure the code works as intended.

Reply:
Yeah, at least the test happy path should be there so we know what we add do what we want